editors/arb-ltip/components/GranteeForm.tsx

```diff
@@ -0,0 +1,274 @@
+import {
+    ArbLtipGranteeState,
+    FundingType,
+} from '../../../document-models/arb-ltip-grantee';
+import { TextInput } from 'document-model-editors';
+import {
+    initPhase,
+    setGranteeMetricsDash,
+    setGranteeName,
+    setGranteeSummary,
+} from '../../../document-models/arb-ltip-grantee/gen/creators';
+import { useCallback, useMemo, useState } from 'react';
+import { maybeToArray } from '../util';
+import DatePicker from 'react-datepicker';
+import { IProps } from '../editor';
+import TagSelector from './TagSelector';
+
+const fundingTypes = [
+    {
+        name: 'EOA',
+        value: 'EOA',
+    },
+    {
+        name: 'Multisig',
+        value: 'Multisig',
+    },
+    {
+        name: '3/5 Multisig',
+        value: 'ThreeofFiveMultisig',
+    },
+    {
+        name: '2/3 Multisig',
+        value: 'TwoofThreeMultisig',
+    },
+];
+
+type GranteeFormProps = ArbLtipGranteeState &
+    Pick<IProps, 'editorContext' | 'dispatch'> & { onClose: () => void };
+
+const GranteeForm = (props: GranteeFormProps) => {
+    const {
+        editorContext,
+        dispatch,
+        onClose,
+        granteeName,
+        metricsDashboardLink,
+        grantSummary,
+        disbursementContractAddress,
+        fundingAddress,
+        fundingType,
+    } = props;
+
+    const [granteeNameLocal, setGranteeNameLocal] = useState(granteeName || '');
+    const [
+        disbursementContractAddressLocal,
+        setDisbursementContractAddressLocal,
+    ] = useState(disbursementContractAddress || '');
+    const [fundingAddressLocal, setFundingAddressLocal] = useState(
+        fundingAddress || '',
+    );
+    const [fundingTypeLocal, setFundingTypeLocal] = useState(
+        maybeToArray(fundingType),
+    );
+    const [grantSummaryLocal, setGrantSummaryLocal] = useState(
+        grantSummary || '',
+    );
+    const [metricsDashboardLinkLocal, setMetricsDashboardLinkLocal] = useState(
+        metricsDashboardLink || '',
+    );
+
+    // 16 weeks
+    const startDate = new Date(2024, 3, 22);
+    const endDate = new Date(2024, 7, 12);
+
+    const onSetGranteeName = useCallback(
+        (name: string) => {
+            setGranteeNameLocal(name);
+            dispatch(setGranteeName({ granteeName: name }));
+        },
+        [dispatch, granteeNameLocal],
+    );
+
+    const onSubmit = useCallback(() => {
+        dispatch(
```

Comment:
At the moment dispatching multiple operations in the same function like this does not work correctly in the connect app, although it does work in storybook. I saw there was some discussion on discord about this and error handling (async dispatch etc.) and it is being worked on.

In the mean time we are using async delays to prevent the stale state problem. See [here](https://github.com/powerhouse-inc/document-model-libs/blob/d5a6154d6be954b7ea9b793d9ec6fefa0972d92f/editors/rwa/transactions.tsx#L172) for an example.

Reply:
Understood. I'll see what we can do. It might make more sense to simply make a single operation that is tailored to this use case.